test-data/unit/check-classes.test

```diff
@@ -7135,24 +7135,20 @@ class B(A):  # E: Final class __main__.B has abstract attributes "foo"
 class C:
     class C1(XX): pass  # E: Name "XX" is not defined
 
-[case testClassScopeImportFunction]
+[case testClassScopeImports]
 class Foo:
-    from mod import foo
+    from mod import plain_function  # E: Unsupported class scoped import
+    from mod import plain_var
 
-reveal_type(Foo.foo)  # N: Revealed type is "def (x: builtins.int, y: builtins.int) -> builtins.int"
-reveal_type(Foo().foo)  # E: Invalid self argument "Foo" to attribute function "foo" with type "Callable[[int, int], int]" \
-                        # N: Revealed type is "def (y: builtins.int) -> builtins.int"
-[file mod.py]
-def foo(x: int, y: int) -> int: ...
+reveal_type(Foo.plain_function)  # N: Revealed type is "Any"
+reveal_type(Foo().plain_function)  # N: Revealed type is "Any"
 
-[case testClassScopeImportVariable]
-class Foo:
-    from mod import foo
+reveal_type(Foo.plain_var)  # N: Revealed type is "builtins.int"
+reveal_type(Foo().plain_var)  # N: Revealed type is "builtins.int"
 
-reveal_type(Foo.foo)  # N: Revealed type is "builtins.int"
-reveal_type(Foo().foo)  # N: Revealed type is "builtins.int"
 [file mod.py]
-foo: int
+def plain_function(x: int, y: int) -> int: ...
+plain_var: int
```

Comment:
What about a variable with a type that is defined only in this module, e.g. a type alias?